backend-shared/config/db/ephemeralPostgres.go

```diff
@@ -0,0 +1,197 @@
+package db
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+type EphemeralDB struct {
+	db            *PostgreSQLDatabaseQueries
+	dbContainerID string
+	dockerNetwork string
+}
+
+// NewEphemeralCreateTestFramework will create a docker container,
+// network, connect to database and populate the schema. This is
+// merely defined for testing purpose
+func NewEphemeralCreateTestFramework() (EphemeralDB, error) {
+	dockerName := "managed-gitops-postgres-test"
+	dockerNetworkcmd := "docker network create %s"
+	uuid := generateUuid()
+	tempDBName := "db-" + uuid
+	tempNetworkName := "gitops-net-" + uuid
+
+	fmt.Println("\nRunning: ", fmt.Sprintf(dockerNetworkcmd, tempNetworkName))
+
+	// #nosec G204
+	dockerNetwork := exec.Command("docker", "network", "create", tempNetworkName)
+	dockerNetworkerr := dockerNetwork.Run()
+	if dockerNetworkerr != nil {
+		return EphemeralDB{}, dockerNetworkerr
+	}
+
+	tempDatabaseDircmd := "mktemp -d -t postgres-XXXXXXXXXX"
+	fmt.Println("\nRunning: ", tempDatabaseDircmd)
+
+	tempDatabaseDir, err_run := exec.Command("mktemp", "-d", "-t", "postgres-XXXXXXXXXX").Output()
+	if err_run != nil {
+		return EphemeralDB{}, err_run
+	}
+
+	dockerContainerIDcmd := `docker run --name ` + dockerName + ` \
+	-v ` + string(tempDatabaseDir) + `:/var/lib/postgresql/data:Z \
+	-e POSTGRES_PASSWORD=gitops \
+	-e POSTGRES_DB=` + tempDBName + ` \
+	-p 6432:5432 \
+	--network ` + tempNetworkName + ` \
+	-d \
+	postgres:13 \
+	-c log_statement='all' \
+	-c log_min_duration_statement=0`
+
+	fmt.Println("\nRunning:", dockerContainerIDcmd)
+
+	var dockerContainerID []byte
+	var errDockerRun error
+	errWait := wait.Poll(5*time.Second, 2*time.Minute, func() (bool, error) {
+		// #nosec G204
+		dockerContainerID, errDockerRun = exec.Command("docker", "run", "--name", dockerName,
+			"-v", string(tempDatabaseDir)+":/var/lib/postgresql/data:Z",
+			"-e", "POSTGRES_PASSWORD=gitops",
+			"-e", "POSTGRES_DB="+tempDBName,
+			"-p", "6432:5432",
+			"--network", "gitops-net-"+uuid,
+			"-d",
+			"postgres:13",
+			"-c", "log_statement=all",
+			"-c", "log_min_duration_statement=0").Output()
+
+		if errDockerRun != nil {
+			log.Fatal(errDockerRun)
+		}
+		if dockerContainerID == nil {
+			return false, errDockerRun
+		}
+		// #nosec G204
+		status, _ := exec.Command("docker", "container", "inspect", "-f", "{{.State.Status}}", string(dockerContainerID)).Output()
+		if string(status) == "running" {
+			return true, nil
+		}
+		// Todo: Once verified removed these print statements in order to maintain code coverage
+		fmt.Println("Docker Container ID: " + string(dockerContainerID))
+		return true, nil
+	})
+	if errWait != nil {
+		return EphemeralDB{}, fmt.Errorf("error in executing docker run command: %s", errWait)
+	}
+
+	dbcmd := "PGPASSWORD=gitops psql -h localhost -d %s -U postgres -p 6432 -c 'select 1'"
```

Comment:
For all cases where we are using psql: Rather than relying on `psql` to be installed on the user's machine, we can instead run psql in a container, similar to what we do with psql.sh: https://github.com/redhat-appstudio/managed-gitops/blob/254e180f93914f9260bc5cfea161bc84909d853c/psql.sh#L7

So for below, that might look something like this:
```go
psqlcmd := exec.Command("docker", "exec", "--rm", "--user", "postgres",  "-e" , "PGPASSWORD=gitops", "-it" "$POSTGRES_CONTAINER", "psql", /* ... */)
```

The `--rm` flag will remove the container as soon as it exists, so there is no need to cleanup after it.

Reply:
I now configured psql command to run from the container side, just a quick note here, correct me if I'm wrong please but docker exec doesn't have rm flag right?